### DISCUSSION TITLE
Root Access on Android

### DISCUSSION THREAD

**Discussion by ghost** [yanivagman/BPFroid#3] [Q&A] [answered]
Do we need a root access on Android (production) to run bpf programs?
Not sure if all process can use bpf syscall

**Comment by yanivagman** [comment, marked as answer]
> Do we need a root access on Android (production) to run bpf programs? Not sure if all process can use bpf syscall

Yes, loading bpf programs into the kernel requires either CAP_SYS_ADMIN on kernels before 5.8 or CAP_BPF for newer kernels. Extra (linux) capabilities such as CAP_PERFMON might be required as well. To run BPFroid, the device needs to be rooted for this reason.
